domain.go

```diff
@@ -36,15 +38,83 @@ type DomainUpdateInput struct {
 }
 
 func (s *DomainId) ChildSystems(client *Client, variables *PayloadVariables) (*SystemConnection, error) {
-	return &SystemConnection{}, nil
+	var q struct {
+		Account struct {
+			Domain struct {
+				ChildSystems SystemConnection `graphql:"childSystems(after: $after, first: $first)"`
+			} `graphql:"domain(input: $domain)"`
+		}
+	}
+	if s.Id == "" {
+		return nil, fmt.Errorf("Unable to get Systems, invalid domain id: '%s'", s.Id)
+	}
+	if variables == nil {
+		variables = client.InitialPageVariablesPointer()
+	}
+
+	(*variables)["domain"] = *NewIdentifier(string(s.Id))
+
+	if err := client.Query(&q, *variables, WithName("DomainChildSystemsList")); err != nil {
+		return nil, err
+	}
+	for q.Account.Domain.ChildSystems.PageInfo.HasNextPage {
+		(*variables)["after"] = q.Account.Domain.ChildSystems.PageInfo.End
+		resp, err := s.ChildSystems(client, variables)
+		if err != nil {
+			return nil, err
+		}
+		q.Account.Domain.ChildSystems.Nodes = append(q.Account.Domain.ChildSystems.Nodes, resp.Nodes...)
+		q.Account.Domain.ChildSystems.PageInfo = resp.PageInfo
+	}
+	q.Account.Domain.ChildSystems.TotalCount = len(q.Account.Domain.ChildSystems.Nodes)
+	return &q.Account.Domain.ChildSystems, nil
 }
 
 func (s *DomainId) Tags(client *Client, variables *PayloadVariables) (*TagConnection, error) {
-	return &TagConnection{}, nil
+	var q struct {
+		Account struct {
+			Domain struct {
+				Tags TagConnection `graphql:"tags(after: $after, first: $first)"`
+			} `graphql:"domain(input: $domain)"`
+		}
+	}
+	if s.Id == "" {
+		return nil, fmt.Errorf("Unable to get Tags, invalid domain id: '%s'", s.Id)
+	}
+	if variables == nil {
+		variables = client.InitialPageVariablesPointer()
+	}
+	(*variables)["domain"] = *NewIdentifier(string(s.Id))
+
+	if err := client.Query(&q, *variables, WithName("DomainTagsList")); err != nil {
+		return nil, err
+	}
+	for q.Account.Domain.Tags.PageInfo.HasNextPage {
+		(*variables)["after"] = q.Account.Domain.Tags.PageInfo.End
+		resp, err := s.Tags(client, variables)
+		if err != nil {
+			return nil, err
+		}
+		q.Account.Domain.Tags.Nodes = append(q.Account.Domain.Tags.Nodes, resp.Nodes...)
+		q.Account.Domain.Tags.PageInfo = resp.PageInfo
+	}
+	q.Account.Domain.Tags.TotalCount = len(q.Account.Domain.Tags.Nodes)
+	return &q.Account.Domain.Tags, nil
 }
 
 func (s *DomainId) AssignSystem(client *Client, systems ...string) error {
-	return nil
+	var m struct {
+		Payload struct {
+			Domain Domain
+			Errors []OpsLevelErrors
+		} `graphql:"domainChildAssign(domain:$domain, childSystems:$childSystems)"`
+	}
+	v := PayloadVariables{
+		"domain":       *NewIdentifier(string(s.Id)),
+		"childSystems": NewIdentifierList(systems),
```

Comment:
This is pedantic but something worth knowing.  In golang they are `Array` which is different from a `List` data structure for a number of reasons.  Array's are static in size you have to allocate more memory to get a bigger array - you define them with a static capacity and then when you "append" to them you are actually creating a whole new array in memory (with increased capacity) and replacing your copy of it - thats why the golang syntax for appending to an array is equals with a method call and you have to pass in a reference the original array - under the hood its making a new array thats bigger, adding all the existing objects and then adding the new object.

Where as `List` is seen as a dynamically "expandable" array which does not exist in golang (unless someone wrote one) but does in other languages.  Under the hood those datatypes provide an interface ontop of array to make them easier to work with.  The main difference being if you took a pointer to an array, then did an `append` on it that original pointer would not have the appended data.

So while I don't care about the name of the function that much - in golang it would be better to be called `NewIdentifierArray` because that means something different then `NewIdentifierList`

Reply:
Cool, thanks for the explanation.  I didn't realize there was a difference and still used to calling everything a list from my python days.  d

Reply:
Quick question about arrays vs slices.  Isn't an `array` always a fixed entity, but `slices` are what can be like an array but also grow and get allocated more storage under the hood?  I think `slice` is a term often used interchangeably with `array`, but they are different if I remember correctly and am reading the difference right?

https://go.dev/blog/slices-intro

Reply:
Ya slices have some attributes close to what `Lists` are like in other languages - While i understand they work differently in golang to array's its just a bit "under the hood" difference with the automatic memory allocation and management that you don't really notice day to day.  for all intents and purposes golang does not truely have "List" built in like other languages.  For example in python you would do
```python
var myList = ["one", "two", "three"]
myList.Append("four")
```
The append method is directly on the type signifiying that it modifies the type in place where in golang you have the `=` operator in the syntax signifying you are making and assigning new memory.

Slices are memory efficient because instead of copying all the memory of the array into a new array they are making a lightweight "pointer like" thing to a chunk of the original array - its slightly weird and not like things in other languages.  I'm pretty sure you also cannot easily tell the difference between a slice and array in golang - i'm not 100% sure if reflection will tell you its an array or slice - where as reflection in other languages will definitely tell you a `List` is not the same type as an `Array`